### PR TITLE
Configure IOptions<EmailServiceOptions> with a Connection String

### PR DESCRIPTION
**Pull request by the-avid-engineer**

### CODE REVIEW

File: src/Postal.AspNetCore/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -17,5 +19,60 @@ public static IServiceCollection AddPostal(this IServiceCollection services)
             services.TryAddScoped<IEmailViewRender, EmailViewRender>();
             return services;
         }
+
+        public static IServiceCollection ConfigurePostal(this IServiceCollection services, string smtpConnectionString)
+        {
+            var match = Regex.Match(smtpConnectionString, @"^smtp:\/\/((?<username>[^:@]+):(?<password>[^:@]+)@)?(?<host>[^:@,\/?#]+)(:(?<port>\d+))?(\?(?<option>[^&]+=[^&]+)(&(?<option>[^&]+=[^&]+))*)?$");
```
Comment (the-avid-engineer) [inline src/Postal.AspNetCore/ServiceCollectionExtensions.cs:25]: This is a watered down version of the mongodb connection string parser

https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L764

**Comment by hermanho** [comment]
@the-avid-engineer Could you please fix the build error?
https://travis-ci.com/hermanho/postal.aspnetcore/builds/144447968

**Comment by hermanho** [comment]
lgtm

File: src/Postal.AspNetCore/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (hermanho) [reply src/Postal.AspNetCore/ServiceCollectionExtensions.cs:25]: 👍

**Pull request merged**